### ISSUE TITLE
Samtools sort clobbreing files

### ISSUE THREAD

**Issue by robsyme** [tseemann/snippy#295]
Hi Torsten

Runninng `snippy 4.3.8`, I noticed that the `snps.log` file included a line that looks like:

```sh
samtools view -h -O SAM /path/to/in.bam \
| samclip --max 10 --ref reference/ref.fa.fai \
| samtools sort -n -l 0 -T $TMPDIR/snippy.385970. --threads 6 -m 1333M \
| samtools fixmate -m - - \
| samtools sort -l 0 -T $TMPDIR/snippy.385970. --threads 6 -m 1333M \
| samtools markdup -T $TMPDIR/snippy.385970. -r -s - - > snps.bam
```

Both invocations of `samtools sort` use the same `-T` temporary file prefix. They are being run in parallel, so I think that they both try to read and write to the same locations. In my stderr, I get lines like:

```
[samclip] Processed 31600000 records...
[samclip] Total SAM records 31620220, removed 850589, allowed 261193, passed 30769631
[samclip] Header contained 4 lines
[samclip] Done.
[bam_sort_core] merging from 6 files and 6 in-memory blocks...
[E::hts_open_format] Failed to open file /lb/scratch/jobs/8039413/snippy.385970..0002.bam
[E::hts_open_format] Failed to open file /lb/scratch/jobs/8039413/snippy.385970..0001.bam
[E::hts_open_format] Failed to open file /lb/scratch/jobs/8039413/snippy.385970..0000.bam
samtools sort: failed to create temporary file "/lb/scratch/jobs/8039413/snippy.385970..0000.bam": File exists
samtools sort: failed to create temporary file "/lb/scratch/jobs/8039413/snippy.385970..0001.bam": File exists
samtools sort: failed to create temporary file "/lb/scratch/jobs/8039413/snippy.385970..0002.bam": File exists
[E::hts_open_format] Failed to open file /lb/scratch/jobs/8039413/snippy.385970..0005.bam
[E::hts_open_format] Failed to open file /lb/scratch/jobs/8039413/snippy.385970..0003.bam
samtools sort: failed to create temporary file "/lb/scratch/jobs/8039413/snippy.385970..0003.bam": File exists
[E::hts_open_format] Failed to open file /lb/scratch/jobs/8039413/snippy.385970..0004.bam
samtools sort: failed to create temporary file "/lb/scratch/jobs/8039413/snippy.385970..0004.bam": File exists
samtools sort: failed to create temporary file "/lb/scratch/jobs/8039413/snippy.385970..0005.bam": File exists
[E::bgzf_flush] File write failed (wrong size)
samtools fixmate: Couldn't write to output file: Broken pipe
[E::bgzf_close] File write failed
[bam_mating] error while closing output file
```

Which looks to me like the `samtools sort` processes are clobbering each others files. 

If snippy gave slightly different temp file prefixes to each process, the files wouldn't be overwritten.

Please let me know if I'm just doing something dumb with snippy.

**Comment by robsyme** [comment]
Ah, I see now that snippy is not to blame. The bug is in our cluster's samtools implementation. Sorry for the confusion.

**Comment by robsyme** [comment]
Ah. In setting up a PR, I see that the problem was raised a few weeks ago in #290  and solved in dc4c7ec9f624148e9ecc62b889e3c1dc7dbdc473. Thanks Torsten!

**Comment by tseemann** [comment]
Yep, fixed that. Hope you didn't waste too much time on it. And thanks for the report!
